src/ansys/fluent/core/launcher/launcher.py

```diff
@@ -576,6 +580,17 @@ def launch_fluent(
                     session.tui.file.read_case(case_filepath)
                 else:
                     session.file.read(file_type="case", file_name=case_filepath)
+            if journal_filepath:
+                if meshing_mode:
+                    session.tui.file.read_journal(journal_filepath)
+                else:
+                    session.file.read_journal(journal_filepath)
+            if case_data_filepath:
+                if not meshing_mode:
+                    session.file.read(file_type="case-dats", file_name=case_data_filepath)
+                else:
+                    raise RuntimeError("Case and data file cannot be read in meshing mode.")
```

Comment:
Is this above analogy correct? @seanpearsonuk, @mkundu1

Reply:
`file_type="case-data"`

Reply:
Done :).